Task3

enginire4ek opened the pull request:
Спроектирована архитектура приложения. Реализован экран получения случайного рецепта.

enginire4ek committed:
Task3
Спроектирована архитектура приложения. Реализован экран получения случайного рецепта.

0xera commented on app/build.gradle:
```diff
@@ -32,8 +32,14 @@ android {
+    implementation project(':feature_fridge:domain')
+    implementation project(':feature_recipe:presentation')
+    implementation project(':feature_recipe:data')
+    implementation project(':feature_recipe:domain')
```
в пока небольшом проекте нет большого смысла дробить модули до такого уровня. можно было обойтись просто :feature:fridge и :feature:recipe, если хочется многомодульность)

0xera commented on core-network-api/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
core_network_api -> core.network.api

0xera commented on core-network-api/src/main/java/com/project/giniatovia/core_network_api/models/AnalyzedInstructions.kt:
```diff
@@ -0,0 +1,8 @@
+import com.google.gson.annotations.SerializedName
+
+data class AnalyzedInstructions(
+    @SerializedName("name"  ) var name  : String?          = null,
```
что-то с форматированием

0xera commented on core-network-api/src/main/java/com/project/giniatovia/core_network_api/interfaces/ProvideInterceptor.kt:
```diff
@@ -0,0 +1,36 @@
+            level = loggingLevel
+        }
+
+        override fun headerInterceptor(): Interceptor {
```
предполагается, что при реализации api/impl. в api нету реализации

0xera commented on feature_fridge/presentation/src/main/java/com/project/giniatovia/presentation/fragments/FridgeFragment.kt:
```diff
@@ -0,0 +1,50 @@
+
+    private lateinit var productAdapter: ProductAdapter
+
+    private var _binding: FragmentFridgeBinding? = null
```
во фрагментах надо биндинг подчищать в  onDestroyView

0xera commented on feature_recipe/domain/build.gradle:
```diff
@@ -0,0 +1,16 @@
+
+dependencies {
+    // RxJava
+    implementation 'io.reactivex.rxjava2:rxandroid:2.1.1'
```
а в core_network это прикидывалось через api и казалось должен быть единый источник зависимостей. Может тогда не надо делать api?

0xera commented on feature_recipe/presentation/src/main/java/com/project/giniatovia/presentation/fragments/RecipeFragment.kt:
```diff
@@ -0,0 +1,57 @@
+
+class RecipeFragment : Fragment() {
+    private lateinit var viewModel: RecipeViewModel
+    lateinit var binding: FragmentRecipeBinding
```
утечет

enginire4ek committed:
Task3
Исправлены замечания. Оставил только feature-модули. Изменил настройку зависимостей. Подчищаю binding :)

0xera commented on app/src/main/java/com/erya/recipesearch/network/api/MakeService.kt:
```diff
@@ -1,4 +1,4 @@
-package com.project.giniatovia.core_network_api.interfaces
+package com.erya.recipesearch.network.api
```
ребрендинг)

0xera commented on app/src/main/java/com/erya/recipesearch/network/api/ProvideOkHttpClientBuilder.kt:
```diff
@@ -19,7 +19,5 @@ interface ProvideOkHttpClientBuilder {
-    class Base(
-        provideInterceptor: ProvideInterceptor,
-    ) : Abstract(provideInterceptor)
+    class Base
```
а есть смысл в таких пустых классах?

enginire4ek replied:
Не заметил, забыл стереть

0xera commented on feature_fridge/data/build.gradle:
```diff
@@ -30,8 +30,7 @@ android {
 dependencies {
-    implementation project(':core-network-impl')
-    implementation project(':core-utils')
+    implementation project(':app')
```
app это модуль приложения? обычно к app все(почти) подключаются, а не наоборот

enginire4ek committed:
Task3
Упрощенная архитектура

enginire4ek committed:
Task3
Коммит оставшихся файлов

0xera commented on app/build.gradle:
```diff
@@ -32,11 +32,24 @@ android {
 
-    implementation 'androidx.core:core-ktx:1.7.0'
-    implementation 'androidx.appcompat:appcompat:1.4.2'
+    api "com.squareup.retrofit2:retrofit:2.9.0"
```
Тут api больше не нужен)

enginire4ek committed:
Task3
api -> implementation
